src/pip/_internal/exceptions.py

```diff
@@ -115,14 +182,12 @@ class MissingPyProjectBuildRequires(DiagnosticPipError):
 
     def __init__(self, *, package: str) -> None:
         super().__init__(
-            message=f"Can not process {package}",
-            context=(
+            message=Text(f"Can not process {package}"),
```

Comment:
Same issue with f strings. I don't know if `package` can contain square brackets, but best to escape to be sure.

Reply:
Actually yes, here it won't matter since it's not processing markup.